src/Symfony/Component/Intl/Tests/TimezonesTest.php

```diff
@@ -562,4 +573,67 @@ public function testForCountryCode()
         $this->assertSame(['Europe/Amsterdam'], Timezones::forCountryCode('NL'));
         $this->assertSame(['Europe/Berlin', 'Europe/Busingen'], Timezones::forCountryCode('DE'));
     }
+
+    /**
+     * @expectedException \Symfony\Component\Intl\Exception\MissingResourceException
+     */
+    public function testForCountryCodeWithUnknownCountry()
+    {
+        Timezones::forCountryCode('foobar');
+    }
+
+    /**
+     * @expectedException \Symfony\Component\Intl\Exception\MissingResourceException
+     */
+    public function testGetCountryCodeWithUnknownTimezone()
+    {
+        Timezones::getCountryCode('foobar');
+    }
+
+    /**
+     * @dataProvider provideTimezones
+     */
+    public function testGetGmtOffsetAvailability(string $timezone)
+    {
+        $this->assertInternalType('int', Timezones::getRawOffset($timezone));
```

Comment:
shouldn't we ensure that with return types instead ?

Reply:
it does :D im on it.